src/append/rolling_file/policy/compound/roll/fixed_window.rs

```diff
@@ -239,6 +246,119 @@ impl Deserialize for FixedWindowRollerDeserializer {
     }
 }
 
+#[cfg(all(unix, not(target_os = "solaris")))]
+mod imp {
+    use std::fs::File;
+    use std::fs::remove_file;
+    use std::path::PathBuf;
+    use std::io::{Error, Result};
+    use std::os::unix::io::AsRawFd;
+    use std::path::Path;
+
+    pub struct RollLockEx {
+        _lock_file: File,
+        lock_path: PathBuf,
+    }
+
+    impl RollLockEx {
+        pub fn new(log_dir: Option<&Path>) -> Result<RollLockEx> {
+            let mut lock_path = PathBuf::new();
+            lock_path.push(log_dir.unwrap_or(Path::new("/tmp")));
+            lock_path.push("roll.lock");
+            let lock_file = File::create(&lock_path)?;
+            let ret = unsafe { libc::flock(lock_file.as_raw_fd(), libc::LOCK_EX | libc::LOCK_NB) };
+            if ret < 0 {
+                Err(Error::last_os_error())
+            } else {
+                Ok(RollLockEx { _lock_file: lock_file, lock_path })
+            }
+        }
+    }
+
+    impl Drop for RollLockEx {
+        fn drop(&mut self) {
+            remove_file(&self.lock_path).ok();
+        }
+    }
+}
+
+#[cfg(target_os = "solaris")]
+mod imp {
+    use std::fs::File;
+    use std::fs::remove_file;
+    use std::path::PathBuf;
+    use std::io::{Error, Result};
+    use std::os::unix::io::AsRawFd;
+    use std::path::Path;
+
+    pub struct RollLockEx {
+        _lock_file: File,
+        lock_path: PathBuf,
+    }
+
+    impl RollLockEx {
+        pub fn new(log_dir: Option<&Path>) -> Result<RollLockEx> {
+            let mut lock_path = PathBuf::new();
+            lock_path.push(log_dir.unwrap_or(Path::new("/tmp")));
+            lock_path.push("roll.lock");
+            let lock_file = File::create(&lock_path)?;
+            let flag = libc::LOCK_EX | libc::LOCK_NB;
```

Comment:
I think i'd need some comments / a walk thru offline to understand and approve this.